### ISSUE TITLE
Error on running unit test using mongodb fixtures

### ISSUE THREAD

**Issue by aminkt** [yiisoft/yii2#16877]
### What steps will reproduce the problem?
My project using `mongodb` as database and when i run my unit test to test  save action of my model then i see that `db` component not found. 

It's true because i'm using `mongodb` and don't need `db` for sql. anyway when i change my settings to rename `mongodb` database setting to `db` and still using  mongodb connection settings i see error that means yii2 are trying to use SQL activerecord methods.

My test class: 
    
    namespace common\tests;
    
    use common\models\Developer;
    use common\tests\fixtures\DeveloperFixture;
    use Faker\Factory;
    
    class DeveloperTest extends \Codeception\Test\Unit
    {
        /**
         * @var \common\tests\UnitTester
         */
        protected $tester;
    
        /**
         * @return array
         */
        public function _fixtures()
        {
            return [
                'user' => [
                    'class' => DeveloperFixture::class,
                    'dataFile' => codecept_data_dir() . 'developer.php'
                ]
            ];
        }
        /**
         * Test to saving user in database.
         * We are using Factory object to create dynamic test cases.
         */
        public function testSaving()
        {
            // use the factory to create a Faker\Generator instance
            $faker = Factory::create();
    
            $developer = new Developer([
                'name' => $faker->name,
                'description' => $faker->sentences
            ]);
    
            $this->assertTrue($developer->save(), 'Developer object saved into database.');
        }
    
    
        protected function _before()
        {
        }
    
        protected function _after()
        {
        }
    }

My commont/config/test-local.php

    <?php
    return yii\helpers\ArrayHelper::merge(
        require __DIR__ . '/main.php',
        require __DIR__ . '/main-local.php',
        require __DIR__ . '/test.php',
        [
            'components' => [
                'mongodb' => require_once ('conf.d/test-db.php')
            ],
        ]
    );

My common/config/conf.d/test-db.php

    <?php
    return
        [
            'class' => '\yii\mongodb\Connection',
            'dsn' => 'mongodb://mongodb/mytestdb', //Using docker container
        ];

My fixture class:

    <?php
    
    namespace common\tests\fixtures;
    
    use yii\mongodb\ActiveFixture;
    
    /**
     * Class Developer
     * Active fixture for using Developer model.
     *
     * @package common\tests\fixtures
     */
    class DeveloperFixture extends ActiveFixture
    {
        public $modelClass = \common\models\Developer::class;
    }

After that i run `vendor/bin/codecept -c core/common run unit models/DeveloperTest
` I see below error:

    ---------
    1) DeveloperTest: Saving
     Test  tests/unit/models/DeveloperTest.php:testSaving
                                                                                        
      [yii\base\InvalidConfigException] Failed to instantiate component or class "db".  
                                                                                        
    #1  /app/vendor/yiisoft/yii2/di/Instance.php:139
    #2  /app/vendor/yiisoft/yii2/di/Container.php:428
    #3  /app/vendor/yiisoft/yii2/di/Container.php:364
    #4  /app/vendor/yiisoft/yii2/di/Container.php:156
    #5  /app/vendor/yiisoft/yii2/di/Instance.php:167
    #6  /app/vendor/yiisoft/yii2/di/Instance.php:137
    #7  /app/vendor/yiisoft/yii2/test/DbFixture.php:41
    #8  /app/vendor/yiisoft/yii2/base/BaseObject.php:109
    #9  yii\base\BaseObject->__construct
    #10 /app/vendor/yiisoft/yii2/di/Container.php:375
    #1  /app/vendor/yiisoft/yii2/di/Container.php:428
    #2  /app/vendor/yiisoft/yii2/di/Container.php:364
    #3  /app/vendor/yiisoft/yii2/di/Container.php:156
    #4  /app/vendor/yiisoft/yii2/di/Instance.php:167
    #5  /app/vendor/yiisoft/yii2/di/Instance.php:137
    #6  /app/vendor/yiisoft/yii2/test/DbFixture.php:41
    #7  /app/vendor/yiisoft/yii2/base/BaseObject.php:109
    #8  yii\base\BaseObject->__construct
    #9  /app/vendor/yiisoft/yii2/di/Container.php:375
    #10 /app/vendor/yiisoft/yii2/di/Container.php:156
    
    --
    
    There was 1 failure:
    
    ---------
    1) DeveloperTest: Saving
     Test  tests/unit/models/DeveloperTest.php:testSaving
    Developer object saved into database.
    Failed asserting that false is true.
    #1  /app/core/common/tests/unit/models/DeveloperTest.php:42

And when i change `mongodb` in test-local.php to `db` i see below error log:

    ---------
    1) DeveloperTest: Saving
     Test  tests/unit/models/DeveloperTest.php:testSaving
                                                                                                       
      [yii\base\UnknownMethodException] Calling unknown method: yii\mongodb\Command::checkIntegrity()  
                                                                                                       
    #1  /app/vendor/yiisoft/yii2/base/BaseObject.php:222
    #2  /app/vendor/yiisoft/yii2/test/InitDbFixture.php:96
    #3  /app/vendor/yiisoft/yii2/test/InitDbFixture.php:78
    #4  /app/vendor/yiisoft/yii2/test/FixtureTrait.php:117
    #5  /app/vendor/symfony/event-dispatcher/EventDispatcher.php:212
    #6  /app/vendor/symfony/event-dispatcher/EventDispatcher.php:44
    
    --
    
    There was 1 failure:
    
    ---------
    1) DeveloperTest: Saving
     Test  tests/unit/models/DeveloperTest.php:testSaving
    Developer object saved into database.
    Failed asserting that false is true.
    #1  /app/core/common/tests/unit/models/DeveloperTest.php:42
    
    ERRORS!
    Tests: 1, Assertions: 1, Errors: 1, Failures: 1.

### What is the expected result?
Use mongodb fixture and load db and pass the test.

### What do you get instead?
Error that i described.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15
| PHP version      | 7.1
| Operating system | linux

**Comment by aminkt** [comment]
Is there any suggestions?

**Comment by samdark** [comment]
No. Maybe @SamMousa has ideas.

**Comment by aminkt** [comment]
@SamMousa 
No. The other one is an issue about mongo db and mysql in codeception.
https://github.com/yiisoft/yii2/issues/16878
This issue forced me to use mysql even i don't need to run codeception test.

**Comment by SamMousa** [comment]
Can't you set the database component to use in the dbfixture class (by creating a subclass?)

**Comment by aminkt** [comment]
Can you explain more?
What do you mean?

**Comment by aminkt** [comment]
@SamMousa 
> Can't you set the database component to use in the dbfixture class (by creating a subclass?)

Mongodb fixtures set mongodb as default db component. my question is why i should have db component for mysql to? I just need mongodb.
